zerver/migrations/0453_backfill_remote_realmauditlog_extradata_to_json_field.py

```diff
@@ -0,0 +1,86 @@
+# Generated by Django 4.0.7 on 2022-09-30 20:30
+
+import ast
+from typing import Type
+
+from django.db import migrations
+from django.db.backends.base.schema import BaseDatabaseSchemaEditor
+from django.db.migrations.state import StateApps
+from django.db.models import JSONField, Model
+from django.db.models.functions import Cast, JSONObject
+
+# This migration is mostly the same as realmauditlog_extra_data_json
+
+OLD_VALUE = "1"
+NEW_VALUE = "2"
+USER_FULL_NAME_CHANGED = 124
+BATCH_SIZE = 5000
+
+
+def do_bulk_backfill_extra_data(
+    audit_log_model: Type[Model], id_lower_bound: int, id_upper_bound: int
+) -> None:
+    # First handle the special case for audit logs with the USER_FULL_NAME_CHANGED event, which stores
+    # the full name not as str(dict()) but a plain str.
+    # Note that we only update the entries where extra_data_json has the default value,
+    # because we do not want to override existing audit log entries with a NEW_VALUE of None for extra_data_json.
+    # We do not need to skip existing entries for other parts of backfilling because we have double-write implemented
+    # so that the backfilled value will still be consistent.
+    audit_log_model.objects.filter(
+        event_type=USER_FULL_NAME_CHANGED,
+        id__range=(id_lower_bound, id_upper_bound),
+        extra_data_json={},
+    ).update(extra_data_json=JSONObject(**{OLD_VALUE: "extra_data", NEW_VALUE: None}))
```

Comment:
How does this work -- do we end up with the literal "extra_data" being stored, or am I missing something in how this works?

(I'm also not sure I understand why `new_value` is `None` here)

Reply:
For `extra_data_json` that is missing and needs to be calculated from `extra_data`, [`JSONObject`](https://docs.djangoproject.com/en/4.2/ref/models/database-functions/#jsonobject) takes the value of `extra_data`, which is the value (for this event, we store the old full name as a regular string), and set it for the key `OLD_VALUE` on the `extra_data_json` field; the `NEW_VALUE` is `None` because we don't have information of the new name from the previous audit log entries.